workflow/controller/operator_test.go

```diff
@@ -6812,19 +6812,98 @@ func TestFailSuspendedAndPendingNodesAfterShutdown(t *testing.T) {
+			taskName:   "task-name",
+			withParam:  `[["arr1", "arr2"]]`,
+			wantedName: "task-name(0:[arr1 arr2])",
+		},
+		{
+			name:       "WithParam: Single string value",
+			taskName:   "task-name",
+			withParam:  `["value"]`,
+			wantedName: "task-name(0:value)",
+		},
+		{
+			name:       "WithParam: Nested object",
+			taskName:   "task-name",
+			withParam:  `[{"nested": {"object": "example"}}]`,
+			wantedName: `task-name(0:nested.object:example,nested:{"object":"example"})`,
+		},
+		{
+			name:     "WithItems: Complex JSON object",
+			taskName: "task-name",
+			withItems: []wfv1.Item{
+				{Value: json.RawMessage(`{"number": 2, "string": "foo", "list": [0, "1"], "json": {"number": 2, "string": "foo", "list": [0, "1"]}}`)},
+			},
+			wantedName: `task-name(0:json.list.0:0,json.list.1:1,json.list:[0,"1"],json.number:2,json.string:foo,json:{"number":2,"string":"foo","list":[0,"1"]},list.0:0,list.1:1,list:[0,"1"],number:2,string:foo)`,
+		},
+		{
+			name:     "WithItems: Array of strings",
+			taskName: "task-name",
+			withItems: []wfv1.Item{
+				{Value: json.RawMessage(`["arr1", "arr2"]`)},
+			},
+			wantedName: "task-name(0:[arr1 arr2])",
+		},
+		{
+			name:     "WithItems: Single string value",
+			taskName: "task-name",
+			withItems: []wfv1.Item{
+				{Value: json.RawMessage(`"value"`)},
+			},
+			wantedName: "task-name(0:value)",
+		},
+		{
+			name:     "WithItems: Nested object",
+			taskName: "task-name",
+			withItems: []wfv1.Item{
+				{Value: json.RawMessage(`{"nested": {"object": "example"}}`)},
+			},
+			wantedName: `task-name(0:nested.object:example,nested:{"object":"example"})`,
+		},
 	}
-	taskBytes, err := json.Marshal(task)
-	require.NoError(t, err)
-	var items []wfv1.Item
-	wfv1.MustUnmarshal([]byte(task.WithParam), &items)
 
-	var newTask wfv1.DAGTask
-	tmpl, _ := template.NewTemplate(string(taskBytes))
-	newTaskName, err := processItem(tmpl, "task-name", 0, items[0], &newTask, "")
-	require.NoError(t, err)
-	assert.Equal(t, `task-name(0:json:{"list":[0,"1"],"number":2,"string":"foo"},list:[0,"1"],number:2,string:foo)`, newTaskName)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var task wfv1.DAGTask
+			var items []wfv1.Item
+
+			if tc.withParam != "" {
+				task.WithParam = tc.withParam
+				err := json.Unmarshal([]byte(tc.withParam), &items)
+				require.NoError(t, err, "Failed to unmarshal withParam")
+				require.Len(t, items, 1, "Expected one item in withParam")
+			} else {
+				task.WithItems = tc.withItems
+				items = tc.withItems
+			}
+
+			taskBytes, err := json.Marshal(task)
+			require.NoError(t, err, "Failed to marshal task")
+
+			tmpl, err := template.NewTemplate(string(taskBytes))
+			require.NoError(t, err, "Failed to create template")
+
+			var newTask wfv1.DAGTask
+			newTaskName, err := processItem(tmpl, tc.taskName, 0, items[0], &newTask, "")
+			require.NoError(t, err, "processItem failed")
+			assert.Equal(t, tc.wantedName, newTaskName)
+		})
+	}
```

Comment:
Hey @weafscast - trying to understand the purpose of `processItem()` function, as I'm not really familiar with it. I'm gathering based on the test that it's the **name** of the Task that's really being set and that's it? Is it the name that makes it so that a user can reference `"{{item.tag}}"`? And is it downstream code that then processes that name to help make sense of "{{item.tag}}"`? 

As for your PR, would an example reference be: `"{{item.tag.subtag}}"`?